cmd/dagger/flags.go

```diff
@@ -218,25 +221,29 @@ func (v *directoryValue) Get(ctx context.Context, dag *dagger.Client, modSrc *da
 		return nil, fmt.Errorf("directory address cannot be empty")
 	}
 
-	// Try parsing as a Git URL
-	parsedGit, err := parseGit(v.String())
-	if err == nil {
-		gitOpts := dagger.GitOpts{
-			KeepGitDir: true,
-		}
-		if authSock, ok := os.LookupEnv("SSH_AUTH_SOCK"); ok {
-			gitOpts.SSHAuthSocket = dag.Host().UnixSocket(authSock)
-		}
-		gitDir := dag.Git(parsedGit.Remote, gitOpts).Branch(parsedGit.Fragment.Ref).Tree()
-		if subdir := parsedGit.Fragment.Subdir; subdir != "" {
-			gitDir = gitDir.Directory(subdir)
+	bk := &osBuildkitClient{}
+	ref, kind := vcs.ConvertToBuildKitRef(ctx, v.String(), bk, vcs.ParseRefStringDir)
```

Comment:
We should be able to remove this function. Instead, we have `parseGit` call `vcs.ParseRef`, and then we can extract host, subpath, etc from there :tada:

We should avoid the dance of converting a parsed string into a raw string to parse it again - that's going to more error prone and difficult to follow :scream: